sdcm/utils/sstable/sstable_utils.py

```diff
@@ -221,3 +215,64 @@ def get_tombstone_date(tombstone_deletion_info: str) -> datetime.datetime:
         full_deletion_date = f'{deletion_date} {deletion_hour}:{deletion_minutes}:{deletion_seconds}'
         full_deletion_date_datetime = datetime.datetime.strptime(full_deletion_date, '%Y-%m-%d %H:%M:%S')
         return full_deletion_date_datetime
+
+
+def is_new_sstable_dump_supported(node) -> bool:
+    """
+    Determines if the new "scylla sstable" dump command is supported based on the Scylla version.
+
+    :param node: A DB node object to provide Scylla version.
+    :return: True if the new "scylla sstable" dump is supported, else False.
+    """
+    if node.is_enterprise:
+        return ComparableScyllaVersion(node.scylla_version) >= "2023.1.3"
+    return ComparableScyllaVersion(node.scylla_version) >= "5.4.0~rc0"
+
+
+def _generate_sstable_dump_command(node, command: str, keyspace: str, table: str) -> str:
+    """
+    Constructs the base command for the "scylla sstable" dump tool.
+
+    :param node: A DB node object to provide Scylla version.
+    :param keyspace: Keyspace name.
+    :param table: Table name.
+    :return: Base command string.
+    """
+    scylla_conf_dir = Path(node.add_install_prefix(SCYLLA_YAML_PATH)).parent
+    return (
+        f'SCYLLA_CONF={scylla_conf_dir} '
+        f'{node.add_install_prefix("/usr/bin/scylla")} sstable {command} '
+        f' --keyspace {keyspace} --table {table} --sstables'
+    )
+
+
+def get_sstable_metadata_dump_command(node, keyspace: str, table: str, debug_log_level: bool = True) -> str:
+    """
+    Constructs the command for dumping sstable metadata using either "sstabledump" or "scylla sstable" tool.
+
+    :param node: A DB node object to provide Scylla version.
+    :param keyspace: Keyspace name.
+    :param table: Table name.
+    :param debug_log_level: print output with a debug level.
+    :return: Command string for metadata dump.
+    """
+    if not is_new_sstable_dump_supported(node):
+        return 'sstabledump'
+    dump_cmd = _generate_sstable_dump_command(node, "dump-scylla-metadata", keyspace, table)
+    if debug_log_level:
+        return f'{dump_cmd} --logger-log-level scylla-sstable=debug'
```

Comment:
just ouf of curiosity, can this exact log level setup be used even for `dump-data`?

Reply:
well, looks like it's less relevant and meant only for debugging failures. 
There's no documentation of it. i only saw this line - 
https://github.com/scylladb/scylladb/blob/1728f9c983f3a6070da10c6c6ac0170a5ae488d0/tools/scylla-sstable.cc#L325